src/np_client/tracking.rs

```diff
@@ -249,58 +229,62 @@ pub struct TrackingDoc {
     free_shipping: Option<String>,
 }
 
-fn deserialize_f32_option<'de, D>(deserializer: D) -> Result<Option<f32>, D::Error>
-where
-    D: serde::de::Deserializer<'de>,
-{
-    let s: serde_json::Value = serde::de::Deserialize::deserialize(deserializer)?;
+pub struct TrackingHandler<'c> {
+    client: &'c NPClient,
+}
 
-    if s.is_number() {
-        return s.to_string().parse::<f32>()
-            .map(|val| Some(val))
-            .map_err(|_| serde::de::Error::unknown_variant(&s.to_string(), &["number or empty string"]))
+impl<'cli> TrackingHandler<'cli> {
+    pub(crate) fn new(client: &'cli NPClient) -> Self {
+        Self { client }
     }
-    if s.is_string() {
-        if s.as_str().unwrap().is_empty() {
-            return Ok(None)
-        } else {
-            return s.as_str().unwrap().parse::<f32>()
-            .map(|val| Some(val))
-            .map_err(|_| serde::de::Error::unknown_variant(&s.to_string(), &["number or empty string"]))
-        }
-        
-    }
-    if s.is_null() {
-        return Ok(None)
+
+    pub fn track_parcel(&self, en: ENumber, phone: String) -> TrackParcelBuilder<'cli> {
+        TrackParcelBuilder::new(self.client, en, phone)
     }
-    Err(serde::de::Error::unknown_variant(&s.to_string(), &["number or empty string"]))
 
 }
 
-fn deserialize_u16_option<'de, D>(deserializer: D) -> Result<Option<u16>, D::Error>
-where
-    D: serde::de::Deserializer<'de>,
-{
-    let s: serde_json::Value = serde::de::Deserialize::deserialize(deserializer)?;
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
+pub struct TrackParcelBuilder<'cli> {
+    #[serde(skip)]
+    client: &'cli NPClient,
 
-    if s.is_number() {
-        return s.to_string().parse::<u16>()
-            .map(|val| Some(val))
-            .map_err(|_| serde::de::Error::unknown_variant(&s.to_string(), &["number or empty string"]))
-    }
-    if s.is_string() {
-        if s.as_str().unwrap().is_empty() {
-            return Ok(None)
-        } else {
-            return s.as_str().unwrap().parse::<u16>()
-            .map(|val| Some(val))
-            .map_err(|_| serde::de::Error::unknown_variant(&s.to_string(), &["number or empty string"]))
+    documents: Vec<Document>,
+}
+
+impl<'cli> TrackParcelBuilder<'cli> {
+    pub fn new(client: &'cli NPClient, en: ENumber, phone_number: String) -> Self {
+        Self {
+            client,
+            documents: vec![Document::new(
+                ENumber::try_from(en).unwrap(),
```

Comment:
If I understand the code correctly, you don't need to do a type cast here because this is the same type. Just pass `en`.